R/utils.R

```diff
@@ -129,31 +129,39 @@ render_dependencies <- function(dependencies, local = TRUE) {
     } else {
       "file"
     }
+    
+    # According to Dash convention, label react and react-dom as originating
+    # in dash_renderer package, even though all three are currently served
+    # up from the DashR package
+    if (dep$name %in% c("react", "react-dom")) {
+      dep$name <- "dash_renderer"
+    }
+
+    # The following lines inject _dash-component-suites into the src tags,
+    # as this is the current Dash convention. The dependency paths cannot
+    # be set solely at component library generation time, since hosted
+    # applications should have the app name injected as well.
+    #
+    # This is essentially analogous to this codeblock on the Python side:
+    # https://github.com/plotly/dash/blob/1249ffbd051bfb5fdbe439612cbec7fa8fff5ab5/dash/dash.py#L207
+    if (is.null(prefix)) {
+      dep[["script"]] <- paste0("/_dash-component-suites/",
+                                dep$name,
+                                "/",
+                                basename(dep[["script"]]))
+    } else {
+      dep[["script"]] <- paste0(sub("/", "", prefix), 
+                                "_dash-component-suites/", 
+                                dep$name,
+                                "/",
+                                basename(dep[["script"]]))
+    }
+    dep$src$file <- ""
```

Comment:
what does this do / why is it necessary? could we add a comment?

Reply:
Yes, I should definitely have added a comment here; this is a bit strange. Starting from

```
dep[["script"]] <- paste0(sub("/", "", prefix), "_dash-component-suites/", 
    dep$name, "/", basename(dep[["script"]]))
```

...then rendering the dependencies with `htmltools`:

`>  htmltools::renderDependencies(list(dep), src, encodeFunc = identity)`

...if we don't blank `dep$src$file`, we obtain

```
<script src="lib/react-dom@15.4.2/_dash-component-suites/dash_renderer/react-dom.min.js"></script>
```

versus

`<script src="/_dash-component-suites/dash_renderer/react-dom.min.js"></script>`

which is what I had thought we wanted.

Reply:
fixed in 0569677c435c928878577db110b4036c9b0b12a8